File: test/test-asm-sub.js
const gb = require("./gbtest")(__filename);

test("test-asm-sub", () => {
  gb.cycles(200);
  expect(gb.gameboy._cpu.a).toBe(0x11);
  expect(gb.gameboy._cpu.d).toBe(0x11);
  expect(gb.gameboy._cpu.e).toBe(0x11);
});

File: test/test-asm-sub.fs
[asm]

main:

\ [HL] A ld,
$22 # A ld, A $8500 ]* ld,
$33 # A ld,
$8500 # HL ld, [HL] A sub,

A D ld,

\ n A sub,
$33 # A ld,
$22 # A sub,

A E ld,

\ B A sub,
$33 # A ld,
$22 # B ld,
B A sub,

begin, halt, repeat,

[endasm]
